iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ArrayTypeReferenceImpl.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.iac.arm.tree.api.NumericLiteral;
+import org.sonar.iac.arm.tree.api.bicep.ArrayTypeReference;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.arm.tree.api.bicep.TypeExpressionAble;
+import org.sonar.iac.arm.tree.impl.AbstractArmTreeImpl;
+import org.sonar.iac.common.api.tree.TextTree;
+import org.sonar.iac.common.api.tree.Tree;
+
+public class ArrayTypeReferenceImpl extends AbstractArmTreeImpl implements ArrayTypeReference {
+  private final TypeExpressionAble type;
+  private final SyntaxToken lBracket;
+  @CheckForNull
+  private final NumericLiteral length;
+  private final SyntaxToken rBracket;
+
+  public ArrayTypeReferenceImpl(TypeExpressionAble type, SyntaxToken lBracket, @Nullable NumericLiteral length, SyntaxToken rBracket) {
+    this.type = type;
+    this.lBracket = lBracket;
+    this.length = length;
+    this.rBracket = rBracket;
+  }
+
+  @Override
+  public String value() {
+    String value = ((TextTree) type).value() + lBracket.value();
+    if (length != null) {
+      value += length.value();
+    }
+    return value + rBracket.value();
+  }
```

Comment:
We usually don't have methods that compute values in that way or that access subelements.
I don't see the added value here, is this method already used in a check?
Usually, we would have a direct getter to `type` and `length,` currently, you have no possibility to access them properly, so you cannot report an issue on length, or you cannot check that `length` is null for example.

Reply:
I thought `ArrayTypeReference` had to extend `TextTree` like `AmbientTypeReference`, removing it will simplify the code, love it 😄

And you're right, totally forgot to generate the getters, got bad habits from using Lombok too much 😔